tripleo/insights/lookup.py

```diff
@@ -95,30 +94,72 @@ def from_interpreters(
         :param outline: Outline shared by the interpreters.
         :param featureset: Gets data from the featureset file.
         :param release: Gets data from the release file.
-        :return: An interpreter for the scenario indicated by the featureset.
-        :raises ValueError: If the featureset points to no scenario file.
+        :return: An interpreter for the scenario defined by the featureset.
+        :raises ValueError: If the featureset points to no environment files.
         """
-        branch_creator = self.tools.branch_creator
-        path_creator = self.tools.path_creator
+        environment = self._get_env_from(outline, featureset, release)
 
-        release = release.get_release_name()
-        scenario = featureset.get_scenario()
-
-        if not scenario:
+        if environment is None:
             raise ValueError(
-                'Featureset has no scenario. '
-                'One is required for the interpreter to be build.'
+                'Featureset has no environments. '
+                'At least one is required for the interpreter to be built.'
             )
 
         return ScenarioInterpreter(
-            self.cache.get(
+            data=environment,
+            overrides=outline.overrides
+        )
+
+    def _get_env_from(
+        self,
+        outline: DeploymentOutline,
+        featureset: FeatureSetInterpreter,
+        release: ReleaseInterpreter
+    ) -> Optional[YAML]:
+        """Gets the environment data for the featureset. This data is
+        generated by aggregating environment files defined by the featureset
+        in their order of appearance.
+
+        :param outline: Outline shared by the interpreters.
+        :param featureset: Gets data from the featureset file.
+        :param release: Gets data from the release file.
+        :return:
+            The environment data for the featureset.
+            'None' if none is defined by the featureset.
+        """
+        envs = featureset.get_environments()
+
+        # Featureset set defines no environment data
+        if not envs:
+            return None
+
+        result = {}
+
+        for env in envs:
+            # Retrieve data from environment file
+            yaml = self.cache.get(
                 Resource(
                     repo=outline.heat,
-                    file=path_creator.create_scenario_path(scenario),
-                    branch=branch_creator.create_release_branch(release)
+                    file=env,
+                    branch=self._get_branch_from(release)
                 )
-            ),
-            overrides=outline.overrides
+            )
+
+            # Merge environment file into the solution
+            result = merge(result, yaml)
```

Comment:
Here is where the data from each of the environment files is combined to form a scenario.